widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.kt

```diff
@@ -0,0 +1,162 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Logger.i(TAG, "Create Call screen")
+        setContentView(R.layout.call_activity)
+        callView = findViewById(R.id.call_view)
+
+        // Legacy company name support
+        Dependencies.sdkConfigurationManager.setLegacyCompanyName(intent.getStringExtra(GliaWidgets.COMPANY_NAME))
+
+        callConfiguration = CallActivityIntentHelper.readConfiguration(this)
+        if (this.intent.hasExtra(GliaWidgets.USE_OVERLAY)) {
+            // Integrator has passed a deprecated GliaWidgets.USE_OVERLAY parameter with Intent
+            // Override bubble configuration with USE_OVERLAY value
+            val useOverlay = this.intent.getBooleanExtra(GliaWidgets.USE_OVERLAY, true)
+            Dependencies.sdkConfigurationManager.setLegacyUseOverlay(useOverlay)
+        }
+
+        if (!callView.shouldShowMediaEngagementView(callConfiguration.isUpgradeToCall)) {
+            finishAndRemoveTask()
+            return
+        }
+
+        callView.setOnTitleUpdatedListener(this::setTitle)
+        callView.setEngagementConfiguration(callConfiguration.engagementConfiguration)
+        callView.setUiTheme(callConfiguration.engagementConfiguration?.runTimeTheme)
+        onBackClickedListener?.also(callView::setOnBackClickedListener)
+
+        // In case the engagement ends, Activity is removed from the device's Recents menu
+        // to avoid app users to accidentally start queueing for another call when they resume
+        // the app from the Recents menu and the app's backstack was empty.
+        callView.setOnEndListener { this.finishAndRemoveTask() }
+
+        callView.setOnMinimizeListener { this.finish() }
+        onNavigateToChatListener?.also(callView::setOnNavigateToChatListener)
+        callView.setOnNavigateToWebBrowserListener(onNavigateToWebBrowserListener)
+
+        if (savedInstanceState == null) {
+            startCall()
+        }
+    }
+
+    override fun onResume() {
+        callView.onResume()
+        super.onResume()
+    }
+
+    override fun onPause() {
+        callView.onPause()
+        super.onPause()
+    }
+
+    override fun onDestroy() {
+        Logger.i(TAG, "Destroy Call screen")
+        onBackClickedListener = null
+        onNavigateToChatListener = null
+        callView.onDestroy()
+        super.onDestroy()
+    }
+
+    override fun onUserInteraction() {
+        super.onUserInteraction()
+        callView.onUserInteraction()
+    }
+
+    private fun startCall() {
+        val engagementConfiguration = callConfiguration.engagementConfiguration!!
+        callView.startCall(
+            engagementConfiguration.companyName!!,
+            engagementConfiguration.queueIds,
+            engagementConfiguration.contextAssetId,
+            engagementConfiguration.screenSharingMode!!,
+            callConfiguration.isUpgradeToCall,
+            callConfiguration.mediaType
+        )
```

Comment:
Could you check if it is possible to remove force null unwraps? I think at least first one should be possible to make cleaner.

Reply:
I prefer to leave this as it is because I will completely change this approach to a more straightforward one during the 'Navigator' implementation.